atd-vze/src/views/Crashes/Crash.js

```diff
@@ -215,60 +218,72 @@ function Crash(props) {
-                    <Table responsive striped hover>
-                      <tbody></tbody>
-                    </Table>
-                  </>
-                ) : (
-                  <>
-                    <CrashEditCoordsMap
-                      data={data.atd_txdot_crashes[0]}
-                      mapGeocoderAddress={mapGeocoderAddress}
-                      crashId={crashId}
-                      refetchCrashData={refetch}
-                      setIsEditingCoords={setIsEditingCoords}
-                    />
-                  </>
-                )}
-              </CardBody>
-            </Card>
-          </div>
+                    "unassigned"}
+                  )
+                  <br />
+                  Geocode Provider:{" "}
+                  {latitude && longitude
+                    ? geocodeMethod.name
+                    : "No Primary Coordinates"}
+                </Col>
+                <Col>
+                  {!isEditingCoords && (
+                    <Button
+                      color="primary"
+                      style={{ float: "right" }}
+                      onClick={e => setIsEditingCoords(!isEditingCoords)}
+                    >
+                      Edit Coordinates
+                    </Button>
+                  )}
+                </Col>
+              </Row>
+            </CardHeader>
+            <CardBody style={{ minHeight: "350px" }}>
+              {(!latitude || !longitude) && (
+                <Alert color="danger">
+                  Crash record is missing latitude and longitude values required
+                  for map display.
+                </Alert>
+              )}
+              {!isEditingCoords && latitude && longitude ? (
+                <>
+                  <CrashMap data={data.atd_txdot_crashes[0]} />
+                </>
+              ) : (
+                <>
+                  <CrashEditCoordsMap
+                    data={data.atd_txdot_crashes[0]}
+                    mapGeocoderAddress={mapGeocoderAddress}
+                    crashId={crashId}
+                    refetchCrashData={refetch}
+                    setIsEditingCoords={setIsEditingCoords}
+                  />
+                </>
+              )}
+            </CardBody>
+          </Card>
         </Col>
-        <Col xs="12" md="6">
-          <CR3Record crashId={crashId} isCr3Stored={cr3StoredFlag === "Y"} isTempRecord={tempRecord} />
+        <Col xs="12" md="6" className="mb-4">
+          <CrashDiagram
+            crashId={crashId}
+            isCr3Stored={cr3StoredFlag === "Y"}
+            isTempRecord={tempRecord}
+            cr3FileMetadata={cr3FileMetadata}
+          />
         </Col>
-        <Col xs="12">
+      </Row>
+      {!!investigatorNarrative ? (
+        <Row>
+          <Col>
+            <CrashNarrative investigatorNarrative={investigatorNarrative} />
+          </Col>
+        </Row>
+      ) : (
+        <div></div>
```

Comment:
Thanks for catching this! This is just a result of me not tying up all my loose ends, sorry about that! When there was not a narrative available I was rendering a narrative card that said narrative unavailable but then I realized that's not necessary because the main card already explains that there's no narrative and I neglected to clear that logic out of the component.